src/libraries/System.Diagnostics.Process/src/System/Diagnostics/ProcessManager.Linux.cs

```diff
@@ -38,39 +38,10 @@ public static ProcessInfo[] GetProcessInfos(string machineName)
         /// <returns>The array of modules.</returns>
         internal static ProcessModuleCollection GetModules(int processId)
         {
-            var modules = new ProcessModuleCollection(0);
-
-            // Process from the parsed maps file each entry representing a module
-            foreach (Interop.procfs.ParsedMapsModule entry in Interop.procfs.ParseMapsModules(processId))
+            ProcessModuleCollection? modules = Interop.procfs.ParseMapsModules(processId);
```

Comment:
suggestion:
```cs
=> Interop.procfs.ParseMapsModules(processId) ?? new (capacity: 0);
```

Reply:
Done, (though bit expensive, as it will call Process.GetExePath instead of early bail. But I guess JIT is smart enough to avoid unnecessary call 🙂)